server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -833,7 +834,7 @@ public IndexResult index(Index index) throws IOException {
                     indexResult = new IndexResult(
                             plan.versionForIndexing, getPrimaryTerm(), plan.seqNoForIndexing, plan.currentNotFoundOrDeleted);
                 }
-                if (index.origin() != Operation.Origin.LOCAL_TRANSLOG_RECOVERY) {
+                if (index.origin().isRemote()) {
```

Comment:
maybe rename this to `isTranslog`? then it will tie directly to what's happening in this code.